File: csharp/pythagorean-triplet/PythagoreanTriplet.cs
using System.Collections.Generic;

public static class PythagoreanTriplet
{
    public static IEnumerable<(int a, int b, int c)> TripletsWithSum(int sum)
    {
        for (var a = 0; a < sum / 3; a++)
        {
            for (var b = 0; b < sum / 2; b++)
            {
                var c = sum - b - a;

                if (a < b && a.Squared() + b.Squared() == c.Squared())
                {
                    yield return (a, b, c);
                }
            }
        }
    }
}

internal static class IntExtensions {
    public static int Squared(this int value) => value * value;
}

File: fsharp/pythagorean-triplet/PythagoreanTriplet.fs
module PythagoreanTriplet

let tripletsWithSum sum =
    let isPythagorean a b c = a < b && b < c && a * a + b * b = c * c

    [ for a in 0 .. sum / 3 do
          for b in 0 .. sum / 2 do
              let c = sum - b - a

              if isPythagorean a b c then
                  yield (a, b, c) ]
